source/MaterialXTest/ShaderValid.cpp

```diff
@@ -753,15 +771,25 @@ bool getTestOptions(const std::string& optionFile, ShaderValidTestOptions& optio
             }
         }
 
+        // Disable render and save of images if not compiled code will be generated
         if (!options.compileCode)
         {
             options.renderImages = false;
             options.saveImages = false;
         }
+        // Disable saveing imsages, if no images are to be produced
         if (!options.renderImages)
         {
             options.saveImages = false;
         }
+
+        // If implementation count check is required, then at a minimum OSL and GLSL
+        // code generation must execute to be able to check implementation usage.
+        if (options.checkImplCount)
```

Comment:
Force run GLSL and OSL otherwise the implementations are not being fully tests so you can't get an accurate count.